src/App.tsx

```diff
@@ -1,204 +1,102 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos } from './api/todos';
+import { FilterStatus, ErrorMessages } from './types';
+import { Todo, USER_ID } from './types';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoList } from './components/TodoList';
+import { filterTodos } from './utils/fiterTodos';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [preparedTodos, setPreparedTodos] = useState<Todo[]>([]);
+  const [activeFilterStatus, setActiveFilterStatus] = useState<FilterStatus>(
+    FilterStatus.All,
+  );
+  const [currentError, setCurrentError] = useState<ErrorMessages | ''>('');
+
+  const handleHideError = (): void => {
+    setCurrentError('');
+  };
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        setIsLoading(true);
+        const data: Todo[] = await getTodos();
+
+        setPreparedTodos(data);
+      } catch (error) {
+        setCurrentError(ErrorMessages.Load);
+        setTimeout(() => {
+          handleHideError();
+        }, 3000);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadTodos();
+  }, [currentError]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const filteredTodos = filterTodos(preparedTodos, activeFilterStatus);
+
+  const handleChangeFilter = (type: FilterStatus) => {
+    setActiveFilterStatus(type);
+  };
+
+  const handleCheckTodo = (id: number) => {
+    setPreparedTodos(prev =>
+      prev.map(todo =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+      ),
+    );
+  };
+
+  const quantityActiveTasks = (): number => {
+    return preparedTodos.filter(todo => !todo.completed).length;
+  };
```

Comment:
what for? just filter it right away?


```suggestion
  const quantityActiveTasks = preparedTodos.filter(todo => !todo.completed).length;
```